tests/tree_construction.rs

```diff
@@ -25,8 +25,7 @@ const DISABLED_CASES: &[&str] = &[
     "<!DOCTYPE html><script></script>  <title>x</title>  </head>",
     "<!DOCTYPE html><html><body><html id=x>",
     r#"<!DOCTYPE html>X</body><html id="x">"#,
-    "<!DOCTYPE html>X</html>",
-    "<!DOCTYPE html>X<p/x/y/z>",
+    "<!DOCTYPE html>X</html>", // Test formatting issue?
```

Comment:
Dropping the override for the `<!DOCTYPE html>X<p/x/y/z>` test case (line 29) and getting 28 to pass in a unit test was what most of this PR is about.